feat: Complete Design Documentation (#27)
mojaloop/business-operations-framework-docs: docs/guide/SettlementBC.md

## Introduction

> The objectives of this implementation is to provide a solution to Settlements that supports the core-settlement operations and the high-level settlement business process functions. This design is an example implementation of a Mojaloop settlement for a specific use case and is by not means the only approach that can be taken. This guide outlines the high-level design and explains the thinking that went into the design.

Reviewer: ```suggestion
The objective of this design is to provide a solution that links the business process functions with core-settlement operations on the switch.

This design is an example implementation of a Mojaloop settlement for a specific use case and is not meant to be comprehensive or cover all scenarios. This guide outlines the high-level design and explains the thinking that went into the design for a specific use-case chosen.```

**Note:** It is important to create the batch settlement object in the way that the settlement is to be completed and finalized.
1. A **Settlement initiation report**  is generated and used to communicated to the settlement bank the requirements of the settlement.
1. **Settlement Finalization & settlement account re-balancing**
This process needs to occur after the settlement bank has applied the settlement changes. In this step the:
   - settlement process is completed and a settlement finalization report has been received from the settlement bank.
   - the net debit participants in the settlement have their position ledgers adjusted.
   - the settlement ledgers are adjusted for all participants to match the transferred amount for the settlement.
   - the settlement ledgers are checked against the real settlement account balances and adjustments processed to ensure that they are aligned.

### Re-balancing function - is not best practice
It is worth noting that the re-balancing function that is defined in the above settlement finalization process is not the preferred or best-practice approach. 

> This approach was chosen because of regularity requirements and limitations of mechanisms available to implement settlement between participants. I.e. it was designed to work on existing in-place financial solutions. Re-balancing has quite a few drawbacks, and is not considered best practice and should be avoided if possible. 

Reviewer: ```suggestion
This approach was chosen because of regulatory requirements and limitations of mechanisms available to implement settlement between participants. I.e. it was designed to work on existing in-place financial solutions. Re-balancing has quite a few drawbacks, and is not considered best practice and should be avoided if possible. 
```

These drawbacks are:
1. Out of sequence re-balancing results in incorrect results. This vulnerability therefore requires a business process and supportive management to enforce.
1. Reconciliation of the Mojaloop Settlement Account and the settlement bank account is difficult and complicated. This is because the re-balancing may not directly reflect the activity in the settlement bank account. The transfer amounts are linked to the timing of when the re-balancing action is applied, and when the reports and statements are generated.

> **Better solution**

Reviewer: ```suggestion
**Recommended solutions**
```
